make_image.sh

```diff
@@ -0,0 +1,231 @@
+	if [ -d "$TEMPDIR" ]; then
+		sudo umount "$TEMPDIR"
+		sudo rmdir "$TEMPDIR"
+	fi
+	if [ -e "$LOOPBACK_DEV" ]; then
+		sudo losetup -d "$LOOPBACK_DEV"
+	fi
+	if [ -f "$TEMPFILE1" ]; then
+		sudo rm -f "$TEMPFILE1"
+	fi
+	if [ -f "$TEMPFILE2" ]; then
+		sudo rm -f "$TEMPFILE2"
+	fi
+	exit 1
+}
+
+echo "**************************************************"
+echo "*** BOOTLOADER: $BAREBOX_IMG"
+echo "*** ERLANG APP: $ERLANG_APP_DIR"
+echo "**************************************************"
+
+echo "*** CREATING BASE IMAGE FILE..."
+dd if=/dev/zero of="$TEMPFILE1" bs=1M count=516
+echo "*** WRITING BOOTLOADER..."
+dd if="$BAREBOX_IMG" of="$TEMPFILE1" conv=notrunc
+echo "*** WRITING PARTITION TABLE..."
+sudo modprobe loop
+sudo losetup "$LOOPBACK_DEV" "$TEMPFILE1"
+sudo sfdisk "$LOOPBACK_DEV" << EOF
+unit: sectors
+
+$LOOPBACK_PART1 : start=8192, size=524288, Id=83
+$LOOPBACK_PART2 : start=532480, size=524288, Id=83
+EOF
+echo "*** FORMATING SYSTEM PARTITION A..."
+sudo partprobe "$LOOPBACK_DEV"
+sudo mkfs.vfat -n "GRISP2A" -s 8 "$LOOPBACK_PART1"
+echo "*** MOUNTING SYSTEM PARTITION A..."
+sudo mount -o uid="$USER" "$LOOPBACK_PART1" "$TEMPDIR"
+
+cd "$ERLANG_APP_DIR"
+echo "*** COMPILING ERLANG APPLICATION..."
+rebar3 compile
+echo "*** DEPLOYING ERLANG APPLICATION TO SYSTEM PARTITION A..."
+rebar3 grisp deploy --destination="$TEMPDIR" --pre-script="true" --post-script="true" --force --relname="$ERLANG_APP_NAME" --relvsn="$ERLANG_APP_VSN"
+
+echo "*** BUILDING RTEMS DEMO..."
+cd "$TOOLCHAIN_DIR/../.."
+make demo
+echo "*** DEPLOYING RTEMS DEMO TO SYSTEM PARTITION A..."
+cp demo/b-imx7/demo.zImage "$TEMPDIR"
+DTBFILE="$( cd "$TEMPDIR"; find . -name imx6ul-grisp2.dtb )"
+if [ ! -f "${TEMPDIR}/${DTBFILE}" ]; then
+	echo "ERROR: Device tree not found in deployed erlang application"
+	false # jump to the cleanup hook
+fi
+cat > "${TEMPDIR}/loader/entries/rtems.conf" <<EOF
+title        GRiSP RTEMS Demo
+version      0.1.0
+linux        /demo.zImage
+devicetree   $DTBFILE
+architecture ARM
+EOF
+
+echo "*** CLEANING UP..."
+sudo umount "$TEMPDIR"
+sudo rmdir "$TEMPDIR"
+sudo losetup -d "$LOOPBACK_DEV"
+
+if [[ $TRUNCATE == 1 ]]; then
+	echo "*** TRUNCATING SYSTEM PARTITION B..."
+	rm -f "$TEMPFILE2"
+	mv "$TEMPFILE1" "$TEMPFILE2"
+	dd if="$TEMPFILE2" of="$TEMPFILE1" bs=1M count=260
+	rm -f "$TEMPFILE2"
+fi
+
+echo "*** COMPRESSING OUTPUT IMAGE..."
+gzip -c "$TEMPFILE1" > "$OUTPUT_FILE"
+rm -f "$TEMPFILE1"
+
+
+echo "**************************************************"
+echo "*** DONE: ${OUTPUT_FILE}"
+echo "**************************************************"
+echo "*** - Copy the image to an SD card:"
+echo "***     OSX:   $ cp \"${OUTPUT_FILE}\" /Volumes/GRISP"
+echo "***            $ diskutil umount /Volumes/GRISP"
+echo "***     Linux: $ cp \"${OUTPUT_FILE}\" /media/$USER/GRISP"
+echo "***            $ umount /media/$USER/GRISP"
+echo "*** - Open a serial console to the GRiSP board"
+echo "***     Linux: $ screen /dev/tty.usbserial-010031 115200"
```

Comment:
`screen` works the same way on macOS too

Reply:
In fact this was for OSX, as I wasn't using Linux for that